### PR TITLE
- replaced for loop by iterator because of concurrent modification in…

### PR DESCRIPTION
**Pull request by BjornWater**
### Prerequisites (Remove this section if you want)

Make sure you...
- Follow the [contribution guidelines](CONTRIBUTING.md)
- Follow the code style (run `./gradlew checkstyleMain`)
- Submit this PR against the `Development` branch.

**Fixed issue:** [Reference the issue number here, or remove if not a fix]

**Changes made:**
Made changes to MapObjects update method . Because you cannot iterate over a for loop and remove from the same collection you are looping over.
So replaced it with a iterator
- List your changes here
- Change 2...
  … MapObjects Update method

### CODE REVIEW

**Comment by FabianTerhorst** [comment]
nice catch

**Comment by vmarchaud** [comment]
If you can do the same for the for loop just upper this one, it would be perfect ;)

**Comment by BjornWater** [comment]
Sure when im done with work will fix the upper loop aswell

**Comment by BjornWater** [comment]
Should be fine to merge now! @vmarchaud

File: src/main/java/com/pokegoapi/api/map/MapObjects.java [hunk 1]
```diff
@@ -184,15 +184,18 @@ public void update(MapObjects other) {
...
-			for (FortData gym : getGyms()) {
-				if (otherGym.getId().equals(gym.getId())) {
-					gyms.remove(gym);
-					break;
-				}
+                
+                
+                for (FortData otherGym : other.getGyms()) {
+                    Iterator<FortData> iterator = gyms.iterator();
+                    while(iterator.hasNext()){
```
Comment (vmarchaud) [inline src/main/java/com/pokegoapi/api/map/MapObjects.java]: i dont think the "while(" pass the checkstyle, there should be a space between, could you please verify with running : 
`gradle checkstyleMain` and correct them if there are error ?

**Comment by FabianTerhorst** [comment]
👍

**Pull request merged**